lib/vector-core/src/config/proxy.rs

```diff
@@ -244,4 +252,87 @@ mod tests {
         assert_eq!(result.http, Some("http://remote.proxy".into()));
         assert_eq!(result.https, Some("https://2.3.4.5:9876".into()));
     }
+
+    #[test]
+    fn build_proxy() {
+        let config = ProxyConfig {
+            http: Some("http://1.2.3.4:5678".into()),
+            https: Some("https://2.3.4.5:9876".into()),
+            ..Default::default()
+        };
+        let http_proxy_result = config
+            .http_proxy()
+            .expect("should not be an error")
+            .expect("should not be None");
+        let https_proxy_result = config
+            .https_proxy()
+            .expect("should not be an error")
+            .expect("should not be None");
+
+        assert_eq!(
+            http_proxy_result.uri().to_string(),
+            Uri::try_from("http://1.2.3.4:5678")
+                .ok()
+                .expect("should not be None")
+                .to_string()
+        );
+        assert_eq!(
+            https_proxy_result.uri().to_string(),
+            Uri::try_from("https://2.3.4.5:9876")
+                .ok()
+                .expect("should not be None")
+                .to_string()
+        );
+    }
+
+    #[test]
+    fn build_proxy_with_basic_authorization() {
+        let config = ProxyConfig {
+            http: Some("http://user:pass@1.2.3.4:5678".into()),
+            https: Some("https://user:pass@2.3.4.5:9876".into()),
+            ..Default::default()
+        };
+        let http_proxy_result = config
+            .http_proxy()
+            .expect("should not be an error")
+            .expect("should not be None");
+        let https_proxy_result = config
+            .https_proxy()
+            .expect("should not be an error")
+            .expect("should not be None");
+        let encoded_header_value = format!("Basic {}", encode("user:pass"));
+
+        assert_eq!(
+            http_proxy_result.uri().to_string(),
+            Uri::try_from("http://user:pass@1.2.3.4:5678")
+                .ok()
+                .expect("should not be None")
+                .to_string()
+        );
+        assert_eq!(
+            http_proxy_result
+                .headers()
+                .get("authorization")
+                .expect("should have authorization header")
+                .to_str()
+                .expect("should be stringified"),
```

Comment:
Could we test for an exact value here and below? This should evaluate to a constant header value.

Reply:
Okay, switched to comparing `HeaderValue` structs instead.